source/backup-cursor.rst

```diff
@@ -0,0 +1,95 @@
+.. _backup-cursor:
+
+============================================================
+$backupCursor and $backupCursorExtended aggregation stages
+============================================================
+
+``$backupCursor`` and ``$backupCursorExtended`` aggregation stages allow developing your custom backup solution: 
+
+* ``$backupCursor`` outputs the list of files and their size to copy.  
+* ``$backupCursorExtended`` outputs the list of WiredTiger transaction log files that have been updated or newly added since the ``$backupCursor`` was first run. Saving these files enables restoring the database to any arbitrary time between the ``$backupCursor`` and ``$backupCursorExtend`` execution times.
+
+They are available in |PSMDB| starting with version 4.4.6-8. 
+
+Percona provides Percona Backup for MongoDB (PBM) – a light-weight open source solution for consistent backups and restores across sharded clusters. PBM relies on these aggregation stages for physical backups and restores. However, if you wish to develop your own backup application, this document describes the ``$backupCursor`` and ``$backupCursorExtended`` aggregation stages.
+
+Usage 
+=============
+
+Run both aggregation stages on one node from each replica set. You can use either the primary or the secondary node. Note that since the secondary node may lag in syncing the data from the primary one, you will have to wait for the exact same time before running the ``$backupCursorExtended``.
+
+Get a list of all files to copy with $backupCursor
+---------------------------------------------------------------
+
+.. code-block:: javascript
+
+    var bkCsr = db.getSiblingDB("admin").aggregate([{$backupCursor: {}}])
+    var bkCsrMetadata = bkCsr.next().metadata
```

Comment:
maybe it's worth removing var in second command so that user actually gets some output, since when cursor returned to a variable using var, it doesn't automatically iterate and it might be confusing)

var bkcsr = db.getSiblingDB("admin").aggregate([{$backupCursor: {}}])
bkCsrMetadata = bkcsr.next().metadata
{
  backupId: new UUID("2fc8ac28-0fbe-4d42-962c-bd1d236f3394"),
  dbpath: '/home/vagrant/percona-server-mongodb-6.0.2-1-x86_64.glibc2.17/nodes/node1/db',
  oplogStart: { ts: Timestamp({ t: 1666641030, i: 1 }), t: Long("-1") },
  oplogEnd: { ts: Timestamp({ t: 1666944139, i: 2 }), t: Long("6") },
  checkpointTimestamp: Timestamp({ t: 1666944086, i: 3 })
}

Reply:
but if cursor is auto iterated then user cannot iterate it to do actual backup because cursor is already exhausted

Reply:
as decided we can remove VAR variable here
bkCsrMetadata = bkcsr.next().metadata

Reply:
agree